### ISSUE TITLE
Support GCC 7.1 for Arangodb

### ISSUE THREAD

**Issue by bissei** [arangodb/arangodb#2823]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [X] 3.1
- [ ] 3.2 Beta1
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [X] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [X] other: Source 3.1.23 & 3.1.24 - Manjaro
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an installation-related issue:
Describe which steps you carried out, what you expected to happen and what actually happened.
Step by step:
wget https://www.arangodb.com/repositories/Source/ArangoDB-3.1.24.tar.gz
or
wget https://www.arangodb.com/repositories/Source/ArangoDB-3.1.23.tar.gz
tar xzf ArangoDB-3.1.23.tar.gz    /   tar xzf ArangoDB-3.1.24.tar.gz
cd ArangoDB-3.1.23    /   cd ArangoDB-3.1.24
mkdir -p build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Release
make

Crashing to following error:
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-format-pedantic’
cc1plus: warning: unrecognized command line option ‘-Wno-gnu-zero-variadic-macro-arguments’
make[3]: *** [tools/gyp/v8_base.target.x64.release.mk:572: /opt/ArangoDB-3.1.24/build/3rdParty/V8/v8/x64.release/obj.target/v8_base/src/heap/heap.o] Error 1
make[2]: *** [3rdParty/V8/CMakeFiles/v8_build.dir/build.make:111: 3rdParty/V8/v8_build-prefix/src/v8_build-stamp/v8_build-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:431: 3rdParty/V8/CMakeFiles/v8_build.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc-multilib/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --enable-libmpx --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release
Thread model: posix
gcc version 7.1.1 20170528 (GCC)

**Comment by dothebart** [comment]
Hi, 
GCC 7.1 is not yet supported. 
We will upgrade V8 in the post 3.2 release phase, up to then you need to choose another compiler - CLANG or GCC 6.

**Comment by bissei** [comment]
Hi,

I installed clang 4.0.1 version and it help for that, but there is still issue. This time with rocksdblib.
I used following cmake command:
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_C_COMPILER=/usr/bin/clang


Scanning dependencies of target rocksdblib
[ 17%] Building CXX object 3rdParty/rocksdb/rocksdb/CMakeFiles/rocksdblib.dir/db/auto_roll_logger.cc.o
In file included from /opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/db/auto_roll_logger.cc:6:
In file included from /opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/db/auto_roll_logger.h:15:
In file included from /opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/port/util_logger.h:18:
In file included from /opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/posix_logger.h:28:
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:75:25: error: no type named 'function' in namespace 'std'
                   std::function<void(void*)> callback);
                   ~~~~~^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:75:33: error: expected ')'
                   std::function<void(void*)> callback);
                                ^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:74:19: note: to match this '('
  void SetCallBack(const std::string point,
                  ^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:102:40: error: no member named 'function' in namespace 'std'
  std::unordered_map<std::string, std::function<void(void*)> > callbacks_;
                                  ~~~~~^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:102:58: error: expected '(' for function-style cast or type construction
  std::unordered_map<std::string, std::function<void(void*)> > callbacks_;
                                                     ~~~~^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:102:59: error: expected expression
  std::unordered_map<std::string, std::function<void(void*)> > callbacks_;
                                                          ^
/opt/ArangoDB-3.1.24/3rdParty/rocksdb/rocksdb/util/sync_point.h:102:62: error: expected member name or ';' after declaration specifiers
  std::unordered_map<std::string, std::function<void(void*)> > callbacks_;
                                                             ^
6 errors generated.
make[2]: *** [3rdParty/rocksdb/rocksdb/CMakeFiles/rocksdblib.dir/build.make:63: 3rdParty/rocksdb/rocksdb/CMakeFiles/rocksdblib.dir/db/auto_roll_logger.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:843: 3rdParty/rocksdb/rocksdb/CMakeFiles/rocksdblib.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

**Comment by jsteemann** [comment]
ArangoDB 3.2, 3.3 and devel now compile with g++7.
There is no g++7 support for ArangoDB 3.1 at the moment.
I am closing this ticket as g++7 support is there for recent versions that are developed most.
